Code/GraphMol/Wrap/Sgroup.cpp

```diff
@@ -0,0 +1,115 @@
+//
+//  Copyright (C) 2019 Greg Landrum
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#define NO_IMPORT_ARRAY
+#include <boost/python.hpp>
+#include <boost/python/suite/indexing/vector_indexing_suite.hpp>
+#include <string>
+
+// ours
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/Sgroup.h>
+#include "props.hpp"
+
+namespace python = boost::python;
+
+namespace RDKit {
+
+namespace {
+python::tuple getMolSGroups(ROMol &mol) {
+  python::list res;
+  std::vector<SGroup> &sgs = getSGroups(mol);
+  for (auto &sg : sgs) {
+    res.append(&sg);
+  }
+  return python::tuple(res);
+}
+void clearMolSGroups(ROMol &mol) {
+  std::vector<SGroup> &sgs = getSGroups(mol);
+  sgs.clear();
+}
+}  // namespace
+
+std::string sGroupClassDoc =
+    "A collection of atoms and bonds with associated properties\n";
+
+struct sgroup_wrap {
+  static void wrap() {
+    python::class_<SGroup, boost::shared_ptr<SGroup>>(
+        "SGroup", sGroupClassDoc.c_str(), python::no_init)
+        .def("GetOwningMol", &SGroup::getOwningMol,
+             "returns the molecule owning this SGroup",
+             python::return_internal_reference<>())
+        .def("GetIndexInMol", &SGroup::getIndexInMol,
+             "returns the index of this SGroup in the owning molecule's list.")
+        .def("GetAtoms", &SGroup::getAtoms,
+             "returns a list of the indices of the atoms in this SGroup",
+             python::return_value_policy<python::copy_const_reference>())
+        .def("GetParentAtoms", &SGroup::getParentAtoms,
+             "returns a list of the indices of the parent atoms in this SGroup",
+             python::return_value_policy<python::copy_const_reference>())
+        .def("GetBonds", &SGroup::getBonds,
+             "returns a list of the indices of the bonds in this SGroup",
+             python::return_value_policy<python::copy_const_reference>())
+        .def("HasProp",

```

Comment:
At some point we should just wrap the Prop base class and derive from that.

Reply:
yep...